app/uk/gov/hmrc/cgtpropertydisposalsfrontend/views/returns/disposaldetails/how_much_did_you_own.scala.html

```diff
@@ -48,89 +47,95 @@
-      assistiveText = messages("unit-input.assistive.percent"),
-      customErrorKey = Some(s"$percentageShareKey$userKey")
-    )),
-    optionHelpText = None
-  )
-)}
-  @selectedIndex = @{
-    form.data.get(key).map({
-      case "0" => 0
-      case "1" => 1
-      case _ => 2
-    })
-  }
-
-@mainTemplate(title = title, userType = request.userType, hasErrors = hasErrors) {
-
-    @backLinkComponent(backLink)
+@title = @{ messages(s"$key$userKey.title") }
+
+@otherHtml = {
+    @govukInput(Input(
+        id = percentageShareKey,
+        name = percentageShareKey,
+        inputType = percentageShareKey,
+        label = Label(
+            content =  HtmlContent(messages(s"$percentageShareKey$userKey.label")),
+        ),
+        errorMessage = if(hasErrors){Some(ErrorMessage(Some(form.error(percentageShareKey) match { case Some(message) => (message.message) case None => ""})))} else None,
+        classes = "govuk-!-width-one-third",
+        spellcheck = Some(false),
+        value = form(percentageShareKey).value,
+        suffix = Some(PrefixOrSuffix(content=HtmlContent("%")))
+    ))
+}
 
-    @if(hasErrors) {
-        @errorSummary(form, None, s"$userKey")
-    }
 
-    @caption(messages("returns.disposal-details.caption"))
+@items = @{List(
+    RadioItem(
+        content = Text(messages(s"returns.$key.Full")),
+        value = Some("0"),
+        conditionalHtml = None,
+        checked = form(key).value.contains("0")
+    ),
+    RadioItem(
+        content = Text(messages(s"returns.$key.Half")),
+        value = Some("1"),
+        conditionalHtml = None,
+        checked = form(key).value.contains("1")
+    ),
+    RadioItem(
+        content = Text(messages(s"returns.$key.Other")),
+        value = Some("2"),
+        conditionalHtml = Some(otherHtml),
+        checked = form(key).value.contains("2")
+    )
+)}
 
-    @formWrapper(controllers.returns.disposaldetails.routes.DisposalDetailsController.howMuchDidYouOwnSubmit(), 'novalidate -> "novalidate") {
+@mainTemplate(pageTitle = title, isWelshTranslationAvailable = true, withSignOutLink = true, backLinkUrl = Some(backLink.url)) {
 
-      @radioGroupConditional(
-        fieldId = key,
-        label = pageHeading(title),
-        selected = selectedIndex,
-        error = form.error(key),
-        options = options,
-        customErrorKey = Some(s"$key$userKey")
-      )
+@if(hasErrors) {
+    @errorSummary(form, None, s"$userKey")
+}
 
-      @submitButton(messages(if(isAmend) "button.continue" else "button.saveAndContinue"))
-    }
+@formHelper(controllers.returns.disposaldetails.routes.DisposalDetailsController.howMuchDidYouOwnSubmit(), 'novalidate -> "novalidate") {
+<fieldset class="govuk-fieldset" id="@key">
+<legend class="govuk-fieldset__legend govuk-fieldset__legend--l">
+    @pageHeading("", Some(messages("returns.disposal-details.caption")))
+</legend>
+    @govukRadios(Radios(
+        fieldset = Some(Fieldset(
+            legend = Some(Legend(
```

Comment:
govukRadios component includes a fieldset and legend, so this will cause a double-wrap